### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by goodwin64**
I got a fresh look at the code and proposed a bunch of little suggestions from my side.

### CODE REVIEW

File: test/parse.js [hunk 1]
```diff
@@ -64,26 +64,26 @@ test('parse query string without a value', t => {
 	t.deepEqual(queryString.parse('a=&a'), {a: ['', null]});
 });
 
-test('return empty object if no qss can be found', t => {
+test('return empty object if no query string search can be found', t => {
```
Comment (goodwin64) [inline test/parse.js]: No definition what "qss" means

File: test/parse.js [hunk 2]
```diff
@@ -222,7 +222,7 @@ test('circuit parse -> stringify', t => {
 });
 
 test('circuit original -> parse - > stringify -> sorted original', t => {
-	const original = 'foo[21474836471]=foo&foo[21474836470]&foo[1]=one&foo[0]=&bat=buz';
+	const original = 'foo[471]=foo&foo[470]&foo[1]=one&foo[0]=&bat=buz';
```
Comment (goodwin64) [inline test/parse.js]: Such big numbers are harder to grasp.

File: index.d.ts [hunk 3]
```diff
@@ -133,7 +133,8 @@ Parse a query string into an object. Leading `?` or `#` are ignored, so you can
 The returned object is created with [`Object.create(null)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create) and thus does not have a `prototype`.
 
 @param query - The query string to parse.
-*/
+@param options - The options describing how to parse the query string.
```
Comment (sindresorhus) [inline index.d.ts]: This is moot. It doesn't say anything that's not already clear.

File: readme.md [hunk 4]
```diff
@@ -144,7 +144,17 @@ queryString.parse('foo=1&foo=2&foo=3');
 Type: `string`\
 Default: `','`
 
-The character used to separate array elements when using `{arrayFormat: 'separator'}`.
+The **single** character used to separate array elements when using `{arrayFormat: 'separator'}`.
```
Comment (sindresorhus) [inline readme.md]: "The character" already means singular. I don't really see the need to point it out.

File: test/parse.js (hunk 1 above)
Comment (sindresorhus) [inline test/parse.js]:
```suggestion
test('return empty object if no query string can be found', t => {
```

File: index.d.ts (hunk 3 above)
Comment (goodwin64) [reply index.d.ts]: It was just absent in the docs and highlighted by static code analyzer :)

Comment (goodwin64) [reply index.d.ts]: Agree, I'll revert it

File: readme.md (hunk 4 above)
Comment (goodwin64) [reply readme.md]: Agree, I'll revert it

File: readme.md [hunk 5]
```diff
@@ -146,6 +146,16 @@ Default: `','`
...
 
+```js
+const queryString = require('query-string');
+
+queryString.parse('foo=bar|baz', {
+	arrayFormat: 'separator',
+	arrayFormatSeparator: '|'
+});
+//=> {foo: ['bar', 'baz']}
+```
```
Comment (sindresorhus) [inline readme.md]: Can you add these examples to https://github.com/sindresorhus/query-string/blob/master/index.d.ts too?

**Comment by sindresorhus** [comment]
Bump

**Pull request closed without merging**